### PR TITLE
[sram_ctrl] Clean up documentation and add blockdiagram

### PR DESCRIPTION
**Pull request by msfschaffner**
This cleans up the documentation, updates the block diagram and fixes a minor RTL error.

### CODE REVIEW

File: hw/ip/sram_ctrl/doc/_index.md [hunk 1]
```diff
@@ -19,12 +19,33 @@ The SRAM controller contains the SRAM data and address scrambling device and pro
...
+
+In [CTR mode](https://en.wikipedia.org/wiki/Block_cipher_mode_of_operation#Counter_(CTR)), the block cipher is used to encrypt a 64bit IV with the scrambling key in order to create a 64bit keystream block that is bitwise XOR'ed with the data in order to transform plaintext into ciphertext and vice versa.
+The IV is assembled by concatenating a nonce with the word address.
+
+If the input data word is smaller than 64bit, the keystream block is truncated to fit the input data width.
+If the input data word is wider than 64bit, the SRAM controller by default instantiates multiple PRINCE primitives in order to create a unique keystream for the full datawidth.
+For area constrained settings, the parameter `ReplicateKeyStream` can be set to 1 in order to replicate the keystream block generated by one single primitive instead of using multiple parallel PRINCE instances (but it should be understood that this lowers the level of security).
+
+Since plain CTR mode does not diffuse the data bits due to the bitwise XOR, the scheme is augmented by passing each individual byte through a two-layer substitution-permutation (S&P) network implemented with the `prim_subst_perm` primitive.
```
Comment (udinator) [inline hw/ip/sram_ctrl/doc/_index.md:41]: just to make sure i understand correctly, nothing happens to the data in between getting output from PRINCE and getting input to the custom S&P network?

**Review by udinator — APPROVED** [review]
LGTM, with a clarifying question

File: hw/ip/sram_ctrl/doc/_index.md (hunk 1 above)
Comment (msfschaffner) [reply hw/ip/sram_ctrl/doc/_index.md:41]: So actually, the data itself is not passed through PRINCE. PRINCE is used to generate a keystream that is XOR'ed over the data (see block diagram).

However, you understood correctly that after the keystream XOR, the data is directly fed into the custom S&P network and then into the memory.

Comment (udinator) [reply hw/ip/sram_ctrl/doc/_index.md:41]: ahh sorry that is what I meant.
got it, thanks!
LGTM

**Comment by msfschaffner** [comment]
Thanks for the review!

**Pull request merged**